src/NUnitFramework/framework/Internal/Execution/SimpleWorkItemDispatcher.cs

```diff
@@ -60,7 +63,16 @@ public void Start(WorkItem topLevelWorkItem)
 
 #if APARTMENT_STATE
             if (topLevelWorkItem.TargetApartment == ApartmentState.STA)
+            {
+#if NETSTANDARD2_0
+                if (!RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+                {
+                    topLevelWorkItem.MarkNotRunnable("Apartment state cannot be set on platforms other than Windows.");
```

Comment:
@CharliePoole Is marking the top-level work item as not runnable going to report a not-runnable error for every test that has an apartment state set?

I feel like this is going to short circuit and not report any test results. Maybe this is the reason macOS and Linux builds are hanging?